src/com/dabomstew/pkrandom/cli/CliRandomizer.java

```diff
@@ -0,0 +1,188 @@
+            System.err.println("WARNING: " + bundle.getString("GUI.starterUnavailable"));
+        }
+        if (settings.isUpdatedFromOldVersion()) {
+            System.err.println("WARNING: " + bundle.getString("GUI.settingsFileOlder"));
+        }
+    }
+
+    public static int invoke(String[] args) {
+        String settingsFilePath = null;
+        String sourceRomFilePath = null;
+        String outputRomFilePath = null;
+        boolean saveAsDirectory = false;
+
+        List<String> allowedFlags = Arrays.asList("-i", "-o", "-s", "-d");
+        for (int i = 0; i < args.length; i++) {
+            if (allowedFlags.contains(args[i])) {
+                switch(args[i]) {
+                    case "-i":
+                        sourceRomFilePath = args[i + 1];
+                        break;
+                    case "-o":
+                        outputRomFilePath = args[i + 1];
+                        break;
+                    case "-s":
+                        settingsFilePath = args[i + 1];
+                        break;
+                    case "-d":
+                        saveAsDirectory = true;
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+        if (settingsFilePath == null || sourceRomFilePath == null || outputRomFilePath == null) {
+            System.err.println("Missing required argument");
+            return 1;
+
+        }
+
+        // now we know we have the right number of args...
+        if (!new File(settingsFilePath).exists()) {
+            System.err.println("Error: could not read settings file");
+            CliRandomizer.printUsage();
+            return 1;
+        }
+
+        // check that everything is readable/writable as appropriate
+        if (!new File(sourceRomFilePath).exists()) {
+            System.err.println("Error: could not read source ROM file");
+            CliRandomizer.printUsage();
+            return 1;
+        }
+
+        // java will return false for a non-existent file, have to check the parent directory
+        if (!new File(outputRomFilePath).getAbsoluteFile().getParentFile().canWrite()) {
+            System.err.println("Error: destination ROM path not writable");
+            CliRandomizer.printUsage();
+            return 1;
+        }
+
+        boolean processResult = CliRandomizer.performDirectRandomization(
+                settingsFilePath,
+                sourceRomFilePath,
+                outputRomFilePath,
+                saveAsDirectory
+        );
+        if (!processResult) {
+            System.err.println("Error: randomization failed");
+            CliRandomizer.printUsage();
+            return 1;
+        }
+        return 0;
+    }
+
+    private static void printUsage() {
+        System.err.println("Usage: java -jar PokeRandoZX.jar cli -s <path to settings file> -i <path to source ROM> -o <path for new ROM> [-d save as directory]");
```

Comment:
You changed this one to print on `stderr`, but I think you missed the other change I suggested here.
```suggestion
        System.err.println("Usage: java -jar PokeRandoZX.jar cli -s <path to settings file> -i <path to source ROM> -o <path for new ROM> [-d]");
        System.err.println("-d: Save 3DS game as directory (LayeredFS)");
```